ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -225,17 +225,20 @@ func resolveHealthCheck(serviceName string, healthCheck *ecs.HealthCheck, ecsPar
 		return nil, err
 	}
 
-	healthCheck.Command = resolveStringSliceResourceOverride(serviceName, healthCheck.Command, healthCheckOverride.Command, "healthcheck command")
-	healthCheck.Interval = resolveIntPointerResourceOverride(serviceName, healthCheck.Interval, healthCheckOverride.Interval, "healthcheck interval")
-	healthCheck.Retries = resolveIntPointerResourceOverride(serviceName, healthCheck.Retries, healthCheckOverride.Retries, "healthcheck retries")
-	healthCheck.Timeout = resolveIntPointerResourceOverride(serviceName, healthCheck.Timeout, healthCheckOverride.Timeout, "healthcheck timeout")
-	healthCheck.StartPeriod = resolveIntPointerResourceOverride(serviceName, healthCheck.StartPeriod, healthCheckOverride.StartPeriod, "healthcheck start_period")
+	if healthCheck != nil {
+		healthCheck.Command = resolveStringSliceResourceOverride(serviceName, healthCheck.Command, healthCheckOverride.Command, "healthcheck command")
+		healthCheck.Interval = resolveIntPointerResourceOverride(serviceName, healthCheck.Interval, healthCheckOverride.Interval, "healthcheck interval")
+		healthCheck.Retries = resolveIntPointerResourceOverride(serviceName, healthCheck.Retries, healthCheckOverride.Retries, "healthcheck retries")
+		healthCheck.Timeout = resolveIntPointerResourceOverride(serviceName, healthCheck.Timeout, healthCheckOverride.Timeout, "healthcheck timeout")
+		healthCheck.StartPeriod = resolveIntPointerResourceOverride(serviceName, healthCheck.StartPeriod, healthCheckOverride.StartPeriod, "healthcheck start_period")
+	} else {
+		healthCheck = healthCheckOverride
+	}
```

Comment:
I was originally going to say this could be simplified if you change the first conditional to `if healthCheck == nil { healthCheck = healthCheckOverride }` ; ideally, you'd even just `return healthCheckOverride, nil` but I realize the Validation comes after. To me, that's maybe a sign that the validation maybe should happen elsewhere but for sake of getting this PR through this is fine for now. Just thought I'd share what I was thinking as far as code design.

Reply:
I just realized that thanks to line 220 there's a case where we aren't doing validation

Reply:
Let me know if you think the new logic can be simplified :)